### DISCUSSION TITLE
Free Node of type "roadServiceArea"

### DISCUSSION THREAD

**Discussion by DLList** [INSPIRE-MIF/helpdesk-validator#806] [General]
Hello,

I have a problem with the ETF-Validator that I'd like to solve here.

I am currently busy developing a tool for transforming street data into INSPIRE Transport Network Structures. I have created several INSPIRE Features and tested them with the Validator. I have a problem with free RoadNodes of type "roadServiceArea".

**My approach for creating INSPIRE-Data**
Creating RoadNodes with the type "roadServiceArea". These objects are geometrically located in the middle of a RoadLink. This means not at the beginning and not at the end.

![RSA](https://user-images.githubusercontent.com/49745407/69551706-71197f80-0f9d-11ea-8dbe-db4f614787bb.png)

**Testing the DataSet with the Validator**
The Validator returns in case of such RoadNodes error message "No free transport nodes".

![RSA_2](https://user-images.githubusercontent.com/49745407/69551866-b76ede80-0f9d-11ea-8829-d8c73f4e6551.png)

**Question**
My question is: Is the fact with the placement of RoadNodes in the middle of RoadLinks (not at their ends) INSPIRE compliant, or is this an error of the validator?

Best regards,
Dawid

**Comment by josemasensio** [comment]
Dear @DLList 

We have been doing some checks with the metadata you provide us on issues #90 and #140 and we have seen that there are not RoadServiceAreas defined, for this reason, the validator interprets that there is a union with a roadServiceArea that does not exist. 
As you can see on [Transport Network TG](https://inspire.ec.europa.eu/documents/Data_Specifications/INSPIRE_DataSpecification_TN_v3.2.pdf) there should be a Service Area defined close enough to RodeNode.
![image](https://user-images.githubusercontent.com/50206416/69734866-03518d00-1130-11ea-9210-0de6f181ce1f.png)

Are you using a different metadata file? Are the Services Areas defined somewhere else?
Regards

**Comment by josemasensio** [comment]
Dear @DLList,

Does the explanation provided was valid for you?

Best regards.

**Comment by DLList** [comment]
Dear @josemasensio,

I am using this data you checked. In this question i didn't ask for FeatureType: RoadServiceArea. 
I asked for RoadNode with the formOfWay-value: 'roadServiceArea'. 
There are two examples for such RoadNodes.

Best regards.

**Comment by josemasensio** [comment]
Dear @DLList,

Regarding your first question,
> Is the fact with the placement of RoadNodes in the middle of RoadLinks (not at their ends) INSPIRE compliant, or is this an error of the validator

INSPIRE only allows RoadNodes in the middle of RoadLink when they are in a roadServiceArea.  For this reason, we were mentioning that there was not a roadServiceArea define on your file.
Secondly, about your last question,
> I asked for RoadNode with the formOfWay-value: 'roadServiceArea'.

There is not any reference on the [TG DataSpecification 3.2](https://inspire.ec.europa.eu/documents/Data_Specifications/INSPIRE_DataSpecification_TN_v3.2.pdf) to the use of `roadServiceArea` as the value of `formOfWayValue`. Here is where the roadServiceArea value is supposed to be defined and which are the accepted values of formOfWayValue:

![image](https://user-images.githubusercontent.com/50206416/70610222-7dd7dd80-1c03-11ea-870f-585eb4900686.png)

Please, let us know what is your opinion about this. 

Best regards.

**Comment by DLList** [comment]
Dear @josemasensio,

As for your first statement,
"INSPIRE only allows RoadNodes in the middle of RoadLink when they are in a roadServiceArea."
Ok, thats new for me. Can I know where concretly it is defined this way?
For the second statement, I'm sorry, I mean't "formOfRoadNode", not "formOfWay".

Best regards.

**Comment by josemasensio** [comment]
Dear @DLList,
 
Thank you for your reply.
 
As you can see at the image below, there are multiple RoadNodes. If we focus on all of them, we can observe that nodes have a connection with other "Entity" i.e.: PortNode, WaterwayNode or another Road. So, on general aspects, we could say that all RoadNodes below are a connections with something.
 
![image](https://user-images.githubusercontent.com/50206416/70780927-f5357a80-1d85-11ea-9959-3238cd6f5b91.png)
 
Regarding Road Transport Network image posted on our previous comment, There are three different types of RoadNodes. Two of them, (roundabout and junction) are connecting different roads. Another example of RoadNode could be a Service Station (real object). The third RoadNode, is connecting the road with a RoadServiceArea which is a surface. [5.4 Application schema Road Transport Networks from Technical Guidance](https://inspire.ec.europa.eu/documents/Data_Specifications/INSPIRE_DataSpecification_TN_v3.2.pdf#page=61&zoom=100,0,122)
 

Regarding your second question: 
> I asked for RoadNode with the formOfRoadNode: 'roadServiceArea'.
 
According to the [TG](https://inspire.ec.europa.eu/documents/Data_Specifications/INSPIRE_DataSpecification_TN_v3.2.pdf#page=74&zoom=100,0,828) a RoadNode is defined as 
`A point spatial object that is used to either represent connectivity between two
road links or to represent a significant spatial object such as a services station or
roundabout.`
 
Therefore, the RoadNode defined, which is raising an error, is being identified as a connection point but is not connected to anything. In order to be compliant needs to be connected to a ServiceArea.
RoadNode have a property which is `formOfRoadNode: 'roadServiceArea'` which means that at this point is the union between the Road and the Service Area, but there is no RoadServiceArea.

In our opinion, the problem is that there is no roadServiceArea defined. Maybe you could try to define a roadServiceArea and run the test again? We will be waiting for your response.    

Please, let us know how can we help you. 

Best regards.

**Comment by DLList** [comment]
Dear @josemasensio,

thank you for answer. Ok, but which association allows to connect a RoadNode to a ServiceArea.

A RoadNode can only be connected with links:
- spokeStart
- spokeEnd

A ServiceArea also has not an association to RoadNode.

Best regards.

**Comment by josemasensio** [comment]
Dear @DLList.

You are right, there is no way to connect directly the RoadNode with the Service area, that is why we are going to use the property of `formOfRoadNode` attribute from RoadNode, which is a description of a RoadNode in the RoadTransportNetwork. 

FormOfRoadNode attribute has some [predefined values](http://inspire.ec.europa.eu/codelist/FormOfRoadNodeValue), which are the following ones:
- enclosed traffic area 
- junction 
- level crossing 
- pseudo node 
- road end 
- road service area 
- roundabout 
- traffic square 

As you can see, one of these values is `roadServiceArea`. This is used to define that there is a ServiceArea, which is what we mentioned on a previous comment:

> The RoadNode defined, which is raising an error, is being identified as a connection point but is not connected to anything. In order to be compliant needs to be connected to a ServiceArea.
RoadNode have a property which is formOfRoadNode: 'roadServiceArea' which means that at this point is the union between the Road and the Service Area, but there is no RoadServiceArea.

What are referring here, is that you have a roadServiceArea attribute on your RoadNode defined in the middle a RoadLink but with no roadServiceArea defined. Maybe you could try to define a roadServiceArea and run the test again?  Please let us know if that works.

Best regards.

**Comment by DLList** [comment]
Hello, all,

I would like to revisit the subject, because it is still not clear for me:

> What are referring here, is that you have a roadServiceArea attribute on your RoadNode defined in the middle a RoadLink but with no roadServiceArea defined. Maybe you could try to define a roadServiceArea and run the test again? Please let us know if that works.
> 

What should the fact that I create a _RoadServiceArea_ bring about? 
- where is it written, that it is allowed if you have _RoadNode_ with the _formOfRoadNode_ attribute value '_RoadServiceArea_'?

How should the _RoadServiceArea_ be connected to my _RoadNode_ with the _formOfRoadNode_ attribute value '_RoadServiceArea_'?
- attributive? => there is no attribute that allows this
- geometrical? => I don't think so

The only connection I make is between my _RoadNode_ and the _RoadServiceType_ (_networkRef_).

So my question is still open:
- Is the fact that the ETF validator sends the error message "The node is not located at the start or end of any link feature in the data set" in this situation correct or not?
If this is correct, it would mean that the _RoadNode_, even if it has the _formOfRoadNode_ value "_RoadServiceArea_", should be connected to a _RoadLink_ (_spokeStart_ or _spokeEnd_). Is this true?

Best regards.
